go/tasks/plugins/array/catalog.go

```diff
@@ -416,13 +422,19 @@ func ConstructOutputReaders(ctx context.Context, dataStore *storage.DataStore, o
 	return outputReaders, nil
 }
 
-func ConstructOutputReader(ctx context.Context, dataStore *storage.DataStore, outputPrefix storage.DataReference,
+func ConstructOutputReader(ctx context.Context, dataStore *storage.DataStore, outputPrefix, baseOutputSandbox storage.DataReference,
 	index int) (io.OutputReader, error) {
-	dataReference, err := dataStore.ConstructReference(ctx, outputPrefix, strconv.Itoa(index))
+	strIndex := strconv.Itoa(index)
+	dataReference, err := dataStore.ConstructReference(ctx, outputPrefix, strIndex)
```

Comment:
i don't follow.  Why are there two data references now?  there's the `dataReference` that's in the original code, and the new `outputSandbox` - what's the difference between the two and why do we need both?

Reply:
so the first reference is for where the metadata is stored -> OutputPath (should have been OutputMetadata). this where essentially where outputs.pb, futures.pb etc
the new one is what is today generated by flytekit, location where data is stored for the execution from the container. This is never read by flytepropeller, but it is better to generate it from propeller